### ISSUE TITLE
Active configuration dump should be in snake_case

### ISSUE THREAD

**Issue by droserasprout** [slskd/slskd#429]
For example, `noVersionCheck` key is printed in the active configuration but not valid in the config (`no_version_check` expected)

**Comment by droserasprout** [comment]
Yeah, many amazing things can be done with YAML, especially with custom parsers and loaders. However, I suggest enforcing a single casing for all options in config and UI. I'm not sure which animal will best suit both the backend and frontend.

**Comment by tomasmorello** [comment]
This is still an issue, where i took the template and i had to figure out that it's actually expecting Snake Case, can i help fixing it ?

**Comment by jpdillingham** [comment]
Sure, that would be great!

The yaml library being used had a version bump somewhere along the way and that may have changed some behavior here.  I'm pretty sure null handling changed, as I reported in #942.
